doc/contributing.md

```diff
@@ -19,14 +12,225 @@ To make a code contribution to the project:
+
+``` javascript
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+// Import our actions from `static/js/ducks/randomString.js`
+import * as Actions from '../ducks/randomString';
+
+
+const RandomString = () => {
+  const value = useSelector((state) => state.randomString);
+  const dispatch = useDispatch();
+
+  return (
+    <div>
+      <div>
+        Back-end random string generator.
+        <br />
+        This is a dummy component for educational purposes.
+        <br />
+        Current value is: {value}
+      </div>
+      <div>
+        <button type="button" onClick={() => dispatch(Actions.fetchRandomString())}>
+          Generate new random string on the back-end.
+        </button>
+      </div>
+    </div>
+  );
+};
+
+export default RandomString;
+```
+
+All of the redux-related logic associated with our new component will live in a separate file in `static/js/ducks`. This is where we'll define our action creators and reducer(s) (`static/js/ducks/randomString.js)`:
+
+``` javascript
+import * as API from '../API';
+
+export const FETCH_RANDOM_STRING = 'skyportal/FETCH_RANDOM_STRING';
+export const FETCH_RANDOM_STRING_OK = 'skyportal/FETCH_RANDOM_STRING_OK';
+const FETCH_RANDOM_STRING_FAIL = 'skyportal/FETCH_RANDOM_STRING_FAIL';
+
+export function fetchRandomString() {
+  return API.GET('/api/random_string', FETCH_RANDOM_STRING);
+}
+
+export default function reducer(state="null", action) {
+  switch (action.type) {
+    case FETCH_RANDOM_STRING_OK: {
+      const { value } = action.data;
+      return value;
+    }
+    case FETCH_RANDOM_STRING_FAIL: {
+      return "null";
+    }
+    default:
+      return state;
+  }
+}
+```
+
+And we'll render our new `RandomString` component at the bottom of our profile page. The `Profile` component is defined in `static/js/components/Profile.jsx`:
+
+``` javascript
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import NewTokenForm from './NewTokenForm';
+import TokenList from './TokenList';
+import RandomString from './RandomString';
+
+
+const Profile = (props) => {
+  const profile = useSelector((state) => state.profile);
+  const groups = useSelector((state) => state.groups.latest);
+  return (
+    <div>
+      <div>
+        Username:
+        {profile.username}
+      </div>
+      <br />
+      <div>
+        User roles:
+        {profile.roles}
+      </div>
+      <br />
+      <NewTokenForm
+        profile={profile}
+        groups={groups}
+      />
+      <br />
+      <TokenList tokens={profile.tokens} />
+      <br />
+      <br />
+      <RandomString />
+    </div>
+  );
+};
+
+export default Profile;
+```
+Note that we've only added a single line (`<RandomString />`), plus a few line breaks, to render our new component within the `Profile` component.
+
+Then we add two lines to our main reducer file to import our new reducer from `randomString.js` and add it to our root reducer (`static/js/reducers.js`):
```

Comment:
Once we merge my reducer branch (I still need to make a PR) this step can go away!